File: leetcode/pascalsTriangle.js
// Leetcode 118: Pascal's Triangle
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
  let triangle = [];
  let firstRow = [1];
  triangle.push(firstRow);
  for (let i = 1; i < numRows; i++) {
    let prevRow = triangle[i - 1];
    let currentRow = [1];
    for (let j = 1; j < i; j++) {
      currentRow.push(prevRow[j - 1] + prevRow[j]);
    }
    currentRow.push(1);
    triangle.push(currentRow);
  }
  return triangle;
};

console.log(generate(5));

File: leetcode/pascals_triangle.py
# Leetcode 118. Pascal's Triangle

from typing import List
def generate(numRows: int) -> List[List[int]]:
  triangle = []

  for i in range(numRows):
    row = [1] * (i + 1)
    for j in range(1, i):
      row[j] = triangle[i - 1][j - 1] + triangle[i - 1][j]
    triangle.append(row)
  
  return triangle

print(generate(5))